geopandas/io/tests/test_file_geom_types_drivers.py

```diff
@@ -302,7 +302,22 @@ def test_to_file_roundtrip(tmpdir, geodataframe, ogr_driver, engine):
                 output_file, driver=driver, engine=engine, **write_kwargs
             )
     else:
-        geodataframe.to_file(output_file, driver=driver, engine=engine, **write_kwargs)
+        if driver == "SQLite" and engine == "pyogrio":
+            try:
+                geodataframe.to_file(
+                    output_file, driver=driver, engine=engine, **write_kwargs
+                )
+            except ValueError as e:
+                if "unrecognized option 'SPATIALITE'" in str(e):
+                    pytest.xfail(
+                        "pyogrio wheels from PyPI do not come with SpatiaLite support. "
+                        f"Error: {e}"
+                    )
+                raise
+        else:
+            geodataframe.to_file(
+                output_file, driver=driver, engine=engine, **write_kwargs
+            )
```

Comment:
To get pyogrio >= 0.7 in all our envs, we need to occasionally install it from PyPI. But it seems that GDAL in those wheels does not have spatialite 

```py
ValueError: unrecognized option 'SPATIALITE' for driver 'SQLite'
```

This is a bit verbose but I couldn't find a better solution as there does not seem to be a way of checking if spatialite is available or not. This way it gets checked if spatialite is available (conda) and xfails if it is not.

Reply:
Just so I'm tracking:
- does GDAL from conda-forge always have SpatialLite?
- do the Fiona wheels from PyPI always include SpatialLite?

Per [pyogrio #330](https://github.com/geopandas/pyogrio/issues/330), It looks like SpatialLite will be available in pyogrio >= 0.8 (not yet released).  So a version guard against pyogrio if installed via pip may be helpful here.

In case it helps, [here](https://github.com/geopandas/pyogrio/blob/main/pyogrio/tests/test_geopandas_io.py#L55) is how we are detecting if SpatialLite is available, by running an SQL query against the data source.

Reply:
> Per [pyogrio #330](https://github.com/geopandas/pyogrio/issues/330), It looks like SpatialLite will be available in pyogrio >= 0.8 (not yet released).


That issue mentions the GDAL small docker images that will now contain a GDAL built with libspatialite, but we are not using those docker images to build our wheels, so that shouldn't matter for the wheels.


> In case it helps, [here](https://github.com/geopandas/pyogrio/blob/main/pyogrio/tests/test_geopandas_io.py#L55) is how we are detecting if SpatialLite is available, by running an SQL query against the data source.

It seems we eventually didn't end up using that in pyogrio itself. The PR where this is being discussed eventually switched to use `INDIRECT_SQL` instead of skipping based on the presence of libspatialite (https://github.com/geopandas/pyogrio/pull/70#discussion_r856874188):

> If gdal is not built with spatialite support in SQLite,
you can use ``sql_dialect="INDIRECT_SQLITE"`` to be able to use spatialite 
functions on native SQLite files like Geopackage. 

---

For our wheels, we indeed don't enable libspatialite in the vcpkg GDAL build. Looking at it again, it seems we _could_ do that . @brendan-ward you commented about that in https://github.com/geopandas/pyogrio/issues/86#issuecomment-1109012589:

> Not sure if this matters, but noting for the record that our vcpkg build does not include `libspatialite` (we don't enable that as an add-on feature); it is LGPL licensed, which likely has license implications for the wheels we create.  I believe Fiona also does not build against `libspatialite`.

In https://github.com/geopandas/pyogrio/pull/161 I first added `libspatialite` feature, but then removed it again before the PR was merged, and I don't directly see any discussion or explanation that explains why I did that .. But there is a bit of discussion later at https://github.com/geopandas/pyogrio/pull/161#issuecomment-1269044332, which again mentions that fiona also doesn't have this.

Anyway, this is a discussion for pyogrio to have to see if we want to include libspatialite support in the wheels. 

Back to this PR, I am actually wondering why we explicitly use `spatialite=True` in the tests instead of using the default sqlite driver options

Reply:
> I am actually wondering why we explicitly use spatialite=True in the tests instead of using the default sqlite driver options

@theroggy do you remember why use explicitly used that when writing the test?

Reply:
Ah, it was added in https://github.com/geopandas/geopandas/pull/2829 just to explicitly test the spatialite option. I think this change is fine then